### PR TITLE
Display error dialog when signing fails

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Display take over diaglog when we get an error during signing.

Display an specific user error message under the form.

## Justify why they are needed

Better error handling.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1193** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1193" target="_blank">Graphite</a> 
    * **PR #1194** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1194" target="_blank">Graphite</a> 
      * **PR #1195** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1195" target="_blank">Graphite</a> 
        * **PR #1196** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1196" target="_blank">Graphite</a> 
          * **PR #1197** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1197" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1197?utm_source=stack-comment).

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1193** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1193" target="_blank">Graphite</a> 
    * **PR #1194** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1194" target="_blank">Graphite</a> 
      * **PR #1195** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1195" target="_blank">Graphite</a> 
        * **PR #1196** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1196" target="_blank">Graphite</a> 
          * **PR #1197** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1197" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1197?utm_source=stack-comment).

File: apps/store/src/components/CheckoutPage/CheckoutPage.tsx [hunk 1]
```diff
@@ -41,75 +43,96 @@ const CheckoutPage = (props: CheckoutPageProps) => {
...
+                name={FormElement.Email}
+                defaultValue={prefilledData.email}
+                required
+              />
+              <Space y={0.5}>
+                <SignButton loading={loading}>
+                  {t('SIGN_BUTTON', { count: cart.entries.length })}
+                </SignButton>
+                <Text size="sm" color="textSecondary" align="center">
+                  {loading
```
Comment (alebedev) [inline apps/store/src/components/CheckoutPage/CheckoutPage.tsx:112]: This ternary is probably too complex, can we just assign `secondaryText` var outside jsx block?

File: apps/store/src/components/CheckoutPage/CheckoutPage.tsx [hunk 2]
```diff
@@ -41,75 +43,96 @@ const CheckoutPage = (props: CheckoutPageProps) => {
...
             </Space>
-          </Space>
-        </form>
-      </Section>
-    </Wrapper>
+          </form>
+        </Section>
+      </Wrapper>
+
+      <FullscreenDialog.Modal
```
Comment (alebedev) [inline apps/store/src/components/CheckoutPage/CheckoutPage.tsx:122]: Do we want error dialog to be fullscreen even on desktop?  I'd expect small modal

**Review by alebedev — APPROVED** [review]
Looks good, some questions inside

I expect we'll have better abstractions for error display when we do it in more places

File: apps/store/src/components/CheckoutPage/CheckoutPage.tsx (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/components/CheckoutPage/CheckoutPage.tsx:122]: Yes, discussed with Zak -- for critical errors are fullscreen: https://www.figma.com/file/8SfzuQEJ4LJ6uKL0l7WKoE/Hedvig.com-%C2%B7-Final-Design?node-id=3087%3A22684&t=ceAK0LEYuojcK2Zr-1

**Comment by robinandeer** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: 2p15yc9KoO3nPlDemyGK.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1197)

**Pull request merged**